packages/block-editor/src/components/block-card/index.js

```diff
@@ -35,12 +30,29 @@ function BlockCard( {
 	const isOffCanvasNavigationEditorEnabled =
 		window?.__experimentalEnableOffCanvasNavigationEditor === true;
 
+	const { parentNavBlockClientId } = useSelect( ( select ) => {
+		const { getSelectedBlockClientId, getBlockParentsByBlockName } =
+			select( blockEditorStore );
+
+		const _selectedBlockClientId = getSelectedBlockClientId();
+
+		return {
+			parentNavBlockClientId: getBlockParentsByBlockName(
+				_selectedBlockClientId,
+				'core/navigation',
```

Comment:
FYI this does couple the block editor to the Nav block (if only for this experiment).

@youknowriad I'm wondering whether we could create a generic React Context (`BlockLibraryContext`?) and use it to wrap the inspector controls within the Nav block. Then we could use the context inside of block editor if it's available to get access to this kind of information without having to hard code it.

Reply:
yeah, that doesn't sound great. What are we trying to do here?

Reply:
So if I'm understanding properly:

  - We're adding a button to select the parent navigation block no matter its place in the parents hierarchy right?

Some questions:

 - Do we want this to be always the case, or it's a temporary thing?
 - Do we want to support the back button for all blocks later down the road?
 - Why not just get the direct parent?

Reply:
> We're adding a button to select the parent navigation block no matter its place in the parents hierarchy right?

Yes. This relates to the offcanvas navigation editor experiment. To quote from the PR description:

> Because when you edit a nested nav item in the offcanvas and then click back you expect to return to the Nav block and not the immediate parent.

----

> Do we want this to be always the case, or it's a temporary thing?

Likely to be _always_ the case. Currently limited to the experiment. 

> Do we want to support the back button for all blocks later down the road?

I'm not sure I can answer that question definitively. Some (@scruffian) have expressed a desire for this. Other's (@draganescu) have been less enamoured with it. 

Subject to further user testing there is always the _possibility_ it might be something we'd want to expose to other/all blocks.

-----

> Why not just get the direct parent?

See above. That's what it does already. But for the offcanvas editor experiment we want it to always go back to the main Navigation block because that is the block where the offcanvas editor is.

Reply:
I think deciding on whether we want to make this generic (for all blocks) (@mtias might have an answer) here is important. Because if we go generic, we can't target the "navigation block" here as this doesn't work, we can target the direct parent (which for me makes more sense)

Reply:
Let's see if we can get clarity. My current understanding (AFAIK) is there isn't an aspiration to introduce a back button to all blocks.

> we can't target the "navigation block" here as this doesn't work, we can target the direct parent (which for me makes more sense)

Can I clarify that you mean we can't do it _like we're doing it in this PR_ (I agree)? Are there other options (beyond [the one I already suggested](https://github.com/WordPress/gutenberg/pull/46037#discussion_r1031542859))?

Reply:
If we really want this specific behavior, I'd start with hard-coding it with a comment. Personally I don't like it though because it feels so natural that a back button in the block inspector card selects the direct parent.

Reply:
The back button should only exist on blocks that control their children (navigation block, contentOnly groups, and maybe Buttons and Social Icons down the road), and should not exist at all for most blocks.

When the back button is present it should always go to the immediate parent that is controlling the children. In the case of navigation, that would be the navigation wrapper.

Reply:
> the immediate parent that is controlling the children. In the case of navigation, that would be the navigation wrapper.

OK so what we need is a means of detecting when a block is being "controlled" by a parent and if it is then get that parent and make it the target of the back button.

Reply:
yeah, which could just be a property of the innerblocks component

Reply:
Or....we could use this

https://developer.wordpress.org/block-editor/reference-guides/data/data-core-block-editor/#areinnerblockscontrolled

Reply:
I'm not sure if that concept is mapping 1:1 with the introduction of list view. How is it being used right now? cc @jorgefilipecosta

Reply:
controlled inner blocks mean the "blocks" are saved to a different entity, so it's the case for: navigation, reusable blocks, template parts but it won't apply to "buttons", "social links"...

Reply:
I've opened [this PR](https://github.com/WordPress/gutenberg/pull/46058) to track the need to make this universal.